src/platforms/mesa/server/kms/platform_symbols.cpp

```diff
@@ -119,12 +131,67 @@ mg::PlatformPriority probe_graphics_platform(
         try
         {
             // Rely on the console handing us a DRM master...
-            console->acquire_device(
+            auto const device_cleanup = console->acquire_device(
                 major(devnum), minor(devnum),
                 std::make_unique<mgc::OneShotDeviceObserver>(tmp_fd)).get();
 
             if (tmp_fd != mir::Fd::invalid)
+            {
+                mgm::helpers::GBMHelper gbm_device{tmp_fd};
+                EGLDisplay dpy;
+                auto dpy_cleanup = mir::raii::paired_calls(
+                    [&dpy, &gbm_device]()
+                    {
+                        dpy = eglGetDisplay(static_cast<EGLNativeDisplayType>(gbm_device.device));
+                    },
+                    [&dpy]()
+                    {
+                        if (dpy != EGL_NO_DISPLAY)
+                        {
+                            eglTerminate(dpy);
+                        }
+                    });
+
+                if (dpy == EGL_NO_DISPLAY)
+                {
+                    mir::log_info(
+                        "Probe failed to create EGL display on device %s: %s",
+                        device.devnode(),
+                        mg::egl_category().message(eglGetError()).c_str());
+                    continue;
+                }
+
+                auto egl_init = mir::raii::paired_calls(
+                    [&dpy]()
+                    {
+                        EGLint major_ver{1}, minor_ver{4};
+                        if (!eglInitialize(dpy, &major_ver, &minor_ver))
+                        {
+                            mir::log_debug("Failed to initialise EGL: %s", mg::egl_category().message(eglGetError()).c_str());
+                            dpy = EGL_NO_DISPLAY;
+                        }
+                    },
+                    [&dpy]()
+                    {
+                        if (dpy != EGL_NO_DISPLAY)
+                        {
+                            eglTerminate(dpy);
+                        }
+                    });
+
+                eglBindAPI(MIR_SERVER_EGL_OPENGL_API);
+
+                if (strncmp(
+                    "llvmpipe",
+                    reinterpret_cast<char const*>(glGetString(GL_RENDERER)),
```

Comment:
Oh wait, this just crashed for me. Bit racey maybe (gdb I cannot reproduce it)
https://pastebin.ubuntu.com/p/TJ5GzCvc9s/
as glGetString(GL_RENDERER) is returning NULL on me sometimes
bors r-